python/pyarrow/dataset.py

```diff
@@ -798,6 +799,18 @@ def write_dataset(data, base_dir, basename_template=None, format=None,
 
             def file_visitor(written_file):
                 visited_paths.append(written_file.path)
+    existing_data_behavior : 'error' | 'overwrite' | 'delete_matching'
```

Comment:
I know "overwrite_or_ignore" would get long, but on the other hand it is also more explicit .. One could misinterpret "overwrite" as "overwrite the full dataset" instead of "overwrite matching files and ignore the others" (one for me "overwrite the full dataset" would mean it deletes all files, whether they would clash or not)

Reply:
I'm fine with overwrite_or_ignore.  I changed it.  I also clarified the docstring for the option a little to be explicit that non-matching files are left alone.

Reply:
Of course this then also makes it inconsistent with R ..